extensions/wikia/CuratedContent/CuratedContentController.class.php

```diff
@@ -400,12 +392,61 @@ function getJsonItem( $titleName, $ns, $pageId ) {
 	 * @return bool
 	 */
 	static function onCuratedContentSave() {
-		self::purgeMethod( 'getList' );
+		global $wgServer;
+
+		$content = F::app()->wg->WikiaCuratedContent;
+
+		( new SquidUpdate( array_unique( array_reduce(
+			$content,
+			function ( $urls, $item ) use ( $wgServer ) {
+				if ( $item[ 'title' ] !== '' && empty( $item[ 'featured' ] ) ) {
+					// Purge section URLs using urlencode() (standard for MediaWiki), which uses implements RFC 1738
+					// https://tools.ietf.org/html/rfc1738#section-2.2 - spaces encoded as `+`.
+					// iOS apps use this variant.
+					$urls[ ] = self::getUrl( 'getList' ) . '&section=' . urlencode( $item[ 'title' ] );
+					// Purge section URLs using rawurlencode(), which uses implements RFC 3986
+					// https://tools.ietf.org/html/rfc3986#section-2.1 - spaces encoded as `%20`.
+					// Android apps use this variant.
+					$urls[ ] = self::getUrl( 'getList' ) . '&section=' . rawurlencode( $item[ 'title' ] );
+					// Purge section URLs using JavaScript encodeURIComponent() compatible standard,
+					// which works almost like rawurlencode(), but does not encode following characters: !'()*
+					// Mercury web app uses this variant - request from Hapi.js to MediaWiki.
+					$urls[ ] = self::getUrl( 'getList' ) . '&section=' . self::encodeURIQueryParam( $item[ 'title' ] );
+					// Mercury web app uses this variant - request from Ember.js to Hapi.js.
+					$urls[ ] =
+						$wgServer .
+						MercuryApiHooks::SERVICE_API_BASE .
+						MercuryApiHooks::SERVICE_API_CURATED_CONTENT .
+						self::encodeURIQueryParam( $item[ 'title' ] );
+				}
+				return $urls;
+			},
+			// Purge all sections list getter URL - no additional params
+			[ self::getUrl( 'getList' ) ]
+		) ) ) )->doUpdate();
```

Comment:
As we're purging four different URLs here maybe it's worth to consider to use surrogate keys here instead.